### PR TITLE
Add coverage for create content-view filter in CLI

### LINKED ISSUE #2459: Automate : content-view filter create in cli
please automate:
$ hammer content-view filter create --help
Usage:
    hammer content-view filter create [OPTIONS]

Options:
 --content-view CONTENT_VIEW_NAME        Content view name
 --content-view-id CONTENT_VIEW_ID       content view numeric identifier
 --description DESCRIPTION               description of the filter
 --inclusion INCLUSION                   specifies if content should be included or excluded, default:
                                         inclusion=false
                                         One of true/false, yes/no, 1/0.
 --name NAME                             name of the filter
 --organization ORGANIZATION_NAME        Organization name to search by
 --organization-id ORGANIZATION_ID       organization ID
 --organization-label ORGANIZATION_LABEL Organization label to search by
 --original-packages ORIGINAL_PACKAGES   Add all packages without Errata to the included/excluded list. (Package
                                         Filter only)
                                         One of true/false, yes/no, 1/0.
 --repositories REPOSITORY_NAMES         Comma separated list of values.
 --repository-ids REPOSITORY_IDS         repository ID
                                         Comma separated list of values.
 --type TYPE                             type of filter (e.g. rpm, package_group, erratum)
 -h, --help                              print help

### PR DESCRIPTION
**Pull request by oshtaier**
Closes #2459
```
nosetests tests/foreman/cli/test_contentviewfilter.py
...S....................S...
----------------------------------------------------------------------
Ran 28 tests in 634.124s
```

Filed BZ:
Bug 1236532 - Cannot get content view filter description using hammer info
https://bugzilla.redhat.com/show_bug.cgi?id=1236532

Updated BZ:
Bug 1228890 - hammer content-view filter create ignores repos if name used instead of ID
https://bugzilla.redhat.com/show_bug.cgi?id=1228890

### CODE REVIEW

**Comment by Ichimonji10** [comment]
ACK

File: tests/foreman/cli/test_contentviewfilter.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+        filter content type and random filter type as a parameters for this
+        filter
+
+        @Feature: Content View Filter
+
+        @Assert: Content view filter created successfully and has correct and
+        expected parameters
+
+        """
+        filter_content_type = gen_choice(['rpm', 'package_group', 'erratum'])
```
Comment (jyejare) [inline tests/foreman/cli/test_contentviewfilter.py]: There is chance of loosing failure here. By generating 1 choice out of this three, we will loose the results for other 2. There is no assurance that all the three types will be covered for that specific run. Better to run loop over here.

File: tests/foreman/cli/test_contentviewfilter.py [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+        filter_type = gen_choice(['true', 'false'])
```
Comment (jyejare) [inline tests/foreman/cli/test_contentviewfilter.py]: same as above.

**Comment by jyejare** [comment]
ACK pending Comments.

**Comment by oshtaier** [comment]
I thought in the same way when I was working on the tests, but in the same time I thought that random values should be enough. Anyway, your vote decided the fate for these tests :smile_cat:

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_contentviewfilter.py
......S......................S...
----------------------------------------------------------------------
Ran 33 tests in 767.956s
```

File: tests/foreman/cli/test_contentviewfilter.py (hunk 1 above)
Comment (Ichimonji10) [inline tests/foreman/cli/test_contentviewfilter.py]: @oshtaier I take it that `test_create_cvf_with_content_types` addresses this comment?

Comment (oshtaier) [inline tests/foreman/cli/test_contentviewfilter.py]: @Ichimonji10 right

File: tests/foreman/cli/test_contentviews.py [hunk 3]
```diff
@@ -662,7 +662,7 @@ def test_associate_view_rh_custom_spin(self):
             'inclusion': 'true',
             'name': name,
         })
-        self.assertEqual(result.return_code, 0)
+        self.assertEqual(result_flt.return_code, 0)
```
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Good catch.

**Comment by elyezer** [comment]
:+1:

**Pull request merged**